### PR TITLE
Simplify pixel log-likelihood.

### PR DESCRIPTION
**Pull request by nbiederbeck**
in the attempt to work on  #1272:

* simplify the log-likelihood analytically
* discard constants and factors which are not needed in minimization

but now the function `ctapipe.image.pixel_likelihood.poisson_likelihood_gaussian` is basically the same as `ctapipe.image.muon.intensity_fitter.negative_log_likelihood` in #1390 and duplicate code is not good.

Maybe do smth like `ctapipe.image.pixel_likelihood.negative_log_likelihood` and use this in the Muon and ImPACT fitting?

### CODE REVIEW

**Comment by MaxNoe** [comment]
Yes, just use this function in the moon code

File: ctapipe/image/pixel_likelihood.py [hunk 1]
```diff
@@ -51,42 +51,62 @@ class PixelLikelihoodError(RuntimeError):
...
+    .. math::
+
+        - \\ln{P} = \\ln{θ} + \\frac{(s - μ)^2}{θ}
+
+
+    References
+    ----------
+    - de Naurois, Rolland
+      'A high performance likelihood reconstruction of gamma-rays
+      for Imaging Atmospheric Cherenkov Telescopes' (2009)
```
Comment (kosack) [inline ctapipe/image/pixel_likelihood.py:88]: could move this to `references.rst`.  See also #1266

**Comment by MaxNoe** [comment]
Could you also apply the same treatment to the other functions and then use the right one in the muon code?

**Comment by nbiederbeck** [comment]
> Could you also apply the same treatment to the other functions and then use the right one in the muon code?

yes

File: ctapipe/image/pixel_likelihood.py (hunk 1 above)
Comment (nbiederbeck) [reply ctapipe/image/pixel_likelihood.py]: done

**Review by kosack — DISMISSED** [review]

**Comment by MaxNoe** [comment]
@nbiederbeck Tests are failing due to an import error. Somewhere, probably it still imports one of the older names.

File: ctapipe/image/pixel_likelihood.py [hunk 2]
```diff
@@ -177,13 +177,15 @@ def neg_log_likelihood(
     approx_mask = prediction > prediction_safety
 
     neg_log_l = 0
-    neg_log_l += neg_log_likelihood_approx(
-        image[approx_mask], prediction[approx_mask], spe_width, pedestal,
-    )
+    if np.sum(approx_mask) > 0:
```
Comment (MaxNoe) [inline ctapipe/image/pixel_likelihood.py:180]: Use `np.count_nonzero` to check how many items are True in a mask. If you just need a single one, use `np.any`

File: ctapipe/image/pixel_likelihood.py [hunk 3]
```diff
@@ -51,214 +46,152 @@ class PixelLikelihoodError(RuntimeError):
...
+    neg_log_l = 0
+    if np.sum(approx_mask) > 0:
+        neg_log_l += neg_log_likelihood_approx(
+            image[approx_mask], prediction[approx_mask], spe_width, pedestal,
         )
-    if np.any(gaus_pix):
-        like[gaus_pix] = poisson_likelihood_gaussian(
-            image[gaus_pix], prediction[gaus_pix], spe_width, ped
+
+    if np.sum(~approx_mask) > 0:
```
Comment (MaxNoe) [inline ctapipe/image/pixel_likelihood.py:185]: This is `not np.any(...)`, right?

Comment (nbiederbeck) [reply ctapipe/image/pixel_likelihood.py:185]: `not np.all(...)` I guess .. will check if there are speed differences between this one and `np.any(~...)`

Comment (nbiederbeck) [reply ctapipe/image/pixel_likelihood.py:185]: well, there is a difference: 
```python
In [24]: mask = np.random.randint(0, 2, size=2000).astype('bool')

In [25]: %timeit np.any(~mask)
4.28 µs ± 22.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [26]: %timeit not np.all(mask)
3.23 µs ± 19.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

but am inclined to keep `np.any(~approx_mask)` due to lower mental overhead

Comment (MaxNoe) [reply ctapipe/image/pixel_likelihood.py:185]: I dont see why having the tilde as unary binary not inside is clearer than not all

Comment (nbiederbeck) [reply ctapipe/image/pixel_likelihood.py:185]: I added the `if` clauses to not give an empty array into the function. I think `any -> not empty (regarding Trues)` is clear. But  `not all -> not empty (regarding Falses)` seems harder than `any (of the other) -> not empty`. Maybe it's just me, I have trouble putting this into written words

Comment (MaxNoe) [reply ctapipe/image/pixel_likelihood.py:185]: This is "hot code" which will be called often by a minimizer, if there is not a huge difference in readability, we should choose the faster code.

**Review by kosack — CHANGES_REQUESTED** [review]
coverage.io mentions a few lines that are not tested in any unit test (see the warnings in the code view), but once those are fixed, I think this is ok.

**Review by kosack — APPROVED** [review]

**Review by maxnoe — APPROVED** [review]

**Pull request merged**